Sources/ReactiveArray.swift

```diff
@@ -3,20 +3,25 @@ import ReactiveSwift
 import Result
 
 public final class ReactiveArray<Element> {
-
 	public typealias Snapshot = ContiguousArray<Element>
 	public typealias Change = Delta<Snapshot, IndexSet>
 
-	public let signal: Signal<Change, NoError>
+	fileprivate let storage: Storage<ContiguousArray<Element>>
```

Comment:
👍

But since we introduced an entity to hold our storage why not have `writeLock` embedded in `Storage` and provide a method to protect any modification? I'm thinking something similar to what `Atomic` provides.

```swift
func modify(_ action: (Value) throws -> Value) rethrows -> Value
```

They are actually closely related like we can see when producer needs to capture both `storage` and `writeLock`.